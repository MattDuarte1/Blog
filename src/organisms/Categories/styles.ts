import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  ContainerSize: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        padding: ${theme.spacings.medium};
      }
    `,
  },
  CategoryBodySize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        div {
          width: 100%;
        }
      }
    `,
  },
};

export const Container = styled.section`
  ${({ theme }) => css`
    display: block;
    padding: ${theme.spacings.xxlarge};
    background-color: ${theme.colors.white};
    width: 100%;
    height: 100%;

    ${BreakPoints.ContainerSize['medium'](theme)}
  `}
`;

export const CategoryHeader = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 2rem;
`;

export const CategoryBody = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    min-width: 100%;
    height: 100%;
    gap: 37px;

    ${BreakPoints.CategoryBodySize['small'](theme)}
  `}
`;
