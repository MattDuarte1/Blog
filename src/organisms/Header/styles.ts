import styled, { css } from 'styled-components';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    height: 86px;
    background-color: ${theme.colors.white};
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.05));

    position: relative;
  `}
`;
