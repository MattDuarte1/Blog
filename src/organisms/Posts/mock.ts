import { IPost } from '@/interfaces/Post';
const Data: IPost[] = [
  {
    id: '92632737',
    title: 'Você conhece Typescript ?',
    content: '',
    description: '',
    metaContent: [],
    createdAt: '2023-02-03T06:06:20+00:00',
    image: {
      url: 'https://www.datocms-assets.com/90479/1674241450-javascript2.svg',
      alt: undefined,
    },
    authorCompact: {
      name: 'Mateus Duarte',
      createdAt: '2023-01-12T06:37:19+00:00',
      image: {
        url: 'https://www.datocms-assets.com/90479/1673505399-67812250_126640048653747_8066872119595701241_n.jpg',
        alt: 'author image',
        title: 'Avatar',
      },
    },
    category: [
      {
        title: 'javascript',
      },
    ],
  },
  {
    id: '92631209',
    title: 'VOCÊ CONHECE JAVASCRIPT ?',
    content: '',
    description: '',
    metaContent: [],
    createdAt: '2023-02-03T06:02:28+00:00',
    image: {
      url: 'https://www.datocms-assets.com/90479/1674241447-javascript1.svg',
      alt: 'alt image',
    },
    authorCompact: {
      name: 'Mateus Duarte',
      createdAt: '2023-01-12T06:37:19+00:00',
      image: {
        url: 'https://www.datocms-assets.com/90479/1673505399-67812250_126640048653747_8066872119595701241_n.jpg',
        alt: 'author image',
        title: 'Avatar',
      },
    },
    category: [
      {
        title: 'javascript',
      },
    ],
  },
  {
    id: '95380505',
    title: 'Como usar Array no Javascript',
    content: '',
    description: '',
    metaContent: [],
    createdAt: '2023-02-03T15:39:27+00:00',
    image: {
      url: 'https://www.datocms-assets.com/90479/1674241454-javascript4.svg',
      alt: 'alt image',
    },
    authorCompact: {
      name: 'Mateus Duarte',
      createdAt: '2023-01-12T06:37:19+00:00',
      image: {
        url: 'https://www.datocms-assets.com/90479/1673505399-67812250_126640048653747_8066872119595701241_n.jpg',
        alt: 'author image',
        title: 'Avatar',
      },
    },
    category: [
      {
        title: 'javascript',
      },
    ],
  },
  {
    id: '95380585',
    title: 'O que é Tailwind Css ?',
    content: '',
    description: '',
    metaContent: [],
    createdAt: '2023-02-03T15:41:53+00:00',
    image: {
      url: 'https://www.datocms-assets.com/90479/1674241443-css2.svg',
      alt: 'image alt',
    },
    authorCompact: {
      name: 'Mateus Duarte',
      createdAt: '2023-01-12T06:37:19+00:00',
      image: {
        url: 'https://www.datocms-assets.com/90479/1673505399-67812250_126640048653747_8066872119595701241_n.jpg',
        alt: 'author image',
        title: 'Avatar',
      },
    },
    category: [
      {
        title: 'tailwind',
      },
    ],
  },
  {
    id: '95380702',
    title: 'React Hooks: o que é e como funciona ?',
    content: '',
    description: '',
    metaContent: [],
    createdAt: '2023-02-03T15:46:51+00:00',
    image: {
      url: 'https://www.datocms-assets.com/90479/1674241456-react.svg',
      alt: undefined,
    },
    authorCompact: {
      name: 'Mateus Duarte',
      createdAt: '2023-01-12T06:37:19+00:00',
      image: {
        url: 'https://www.datocms-assets.com/90479/1673505399-67812250_126640048653747_8066872119595701241_n.jpg',
        alt: 'author image',
        title: 'Avatar',
      },
    },
    category: [
      {
        title: 'react',
      },
    ],
  },
  {
    id: '95380751',
    title: 'Vue JS: O que é, como funciona ?',
    content: '',
    description: '',
    metaContent: [],
    createdAt: '2023-02-03T15:48:45+00:00',
    image: {
      url: 'https://www.datocms-assets.com/90479/1674243496-vuejs.svg',
      alt: undefined,
    },
    authorCompact: {
      name: 'Mateus Duarte',
      createdAt: '2023-01-12T06:37:19+00:00',
      image: {
        url: 'https://www.datocms-assets.com/90479/1673505399-67812250_126640048653747_8066872119595701241_n.jpg',
        alt: 'author image',
        title: 'Avatar',
      },
    },
    category: [
      {
        title: 'vuejs',
      },
    ],
  },
  {
    id: '92697360',
    title: 'o que é CSS3?',
    content: '',
    description: '',
    metaContent: [],
    createdAt: '2023-02-03T09:07:58+00:00',
    image: {
      url: 'https://www.datocms-assets.com/90479/1674241439-css.svg',
      alt: 'css3',
    },
    authorCompact: {
      name: 'Mateus Duarte',
      createdAt: '2023-01-12T06:37:19+00:00',
      image: {
        url: 'https://www.datocms-assets.com/90479/1673505399-67812250_126640048653747_8066872119595701241_n.jpg',
        alt: 'author image',
        title: 'Avatar',
      },
    },
    category: [
      {
        title: 'css',
      },
    ],
  },
];

export default Data;
