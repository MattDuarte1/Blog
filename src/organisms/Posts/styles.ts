import styled, { css, DefaultTheme } from 'styled-components';

const MediaBreakPoints = {
  small: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      padding: 3.5rem;
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 4.4rem;
    background-color: ${theme.colors.white};
    ${MediaBreakPoints.small(theme)}
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    margin-top: 1.4rem;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(26rem, 2fr));
    gap: ${theme.spacings.large};

    @media (max-width: 1065px) {
      grid-template-columns: repeat(auto-fill, minmax(27rem, 1fr));
    }

    @media (max-width: 705px) {
      justify-content: center;
      grid-template-columns: 0fr;
    }
  `}
`;
