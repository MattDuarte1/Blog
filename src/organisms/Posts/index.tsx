import { useRouter } from 'next/router';
import { IPost } from '@/interfaces/Post';
import Heading from '@/atoms/Heading';
import PostCard from '@/molecules/PostCard';
import * as Styled from './styles';
import { forwardRef } from 'react';

interface PostSectionProps {
  data: IPost[];
}

const Posts = forwardRef<HTMLDivElement, PostSectionProps>(
  ({ data, ...props }, ref) => {
    const router = useRouter();
    const handleClick = (e: React.MouseEvent<HTMLDivElement>) => {
      e.preventDefault();
      router.push({ pathname: `/post/${e.currentTarget.id}` });
    };
    return (
      <Styled.Container id="posts" ref={ref} {...props}>
        <Heading size="large">Todos os Posts</Heading>
        <Styled.Content>
          {data.map((item) => (
            <PostCard key={item.id} {...item} onClick={handleClick} />
          ))}
        </Styled.Content>
      </Styled.Container>
    );
  },
);

Posts.displayName = 'Posts Component';

export default Posts;
