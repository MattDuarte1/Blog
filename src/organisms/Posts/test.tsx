import Posts from '.';
import { renderTheme } from '@/styles/render-theme';
import PostsMocked from './mock';
import mockRouter from 'next-router-mock';
import { act, fireEvent, screen, waitFor } from '@testing-library/react';
import { theme } from '@/styles/theme';

jest.mock('next/router', () => require('next-router-mock'));

describe('<PostSection>', () => {
  it('Should render a Posts component', async () => {
    let container: HTMLElement;
    await act(async () => {
      const result = renderTheme(<Posts data={PostsMocked} ref={null} />);

      container = result.container;
    });

    waitFor(() => {
      expect(container.firstElementChild).toBeInTheDocument();
      expect(container.firstElementChild).toHaveAttribute('id', 'posts');
      expect(container.firstElementChild).toHaveStyle({
        padding: '4.4rem',
        'background-color': theme.colors.white,
      });
      expect(container.firstElementChild).toHaveStyleRule('padding', '3.5rem', {
        media: theme.media.lteSmall,
      });
    });
  });

  it('Should navigate to post[id] when clicked in PostCard', async () => {
    await act(async () => {
      renderTheme(<Posts data={PostsMocked} />);
    });
    const firstPost = screen.getAllByRole('article')[1];

    fireEvent.click(firstPost);
    expect(mockRouter).toMatchObject({
      asPath: `/post/${firstPost.getAttribute('id')}`,
      pathname: `/post/${firstPost.getAttribute('id')}`,
    });
  });

  it('Should match snapshot component', async () => {
    let container: HTMLElement;
    await act(async () => {
      const result = renderTheme(<Posts data={PostsMocked} />);

      container = result.container;
    });

    waitFor(() => expect(container.firstElementChild).toMatchSnapshot());
  });
});
