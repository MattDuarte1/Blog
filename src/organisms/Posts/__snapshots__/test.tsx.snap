// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostSection> Should match snapshot component 1`] = `
.c1 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 3.576rem;
  text-transform: none;
  color: #2B2C34;
}

.c4 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 1.617rem;
  text-transform: none;
  color: #2B2C34;
}

.c10 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 2.156rem;
  text-transform: none;
  color: #2B2C34;
}

.c3 {
  width: 300px;
  height: 425px;
  padding: 1.4rem;
  background: #FFFFFE;
  position: relative;
  box-shadow: 4px 6px 13px rgba(0,0,0,0.1);
  border-radius: 12px;
  text-align: start;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  cursor: pointer;
}

.c3 > h1 {
  margin-left: 1rem;
}

.c3 img {
  margin-bottom: 2rem;
}

.c3:hover {
  -webkit-transform: scale(1.02);
  -ms-transform: scale(1.02);
  transform: scale(1.02);
  -webkit-filter: drop-shadow(0 1.5px 2px rgba(0,0,0,0.2));
  filter: drop-shadow(0 1.5px 2px rgba(0,0,0,0.2));
}

.c5 {
  position: absolute;
  padding: 0 1.4rem;
  width: 100%;
  height: 57px;
  left: 0px;
  top: 348px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #2B2C34;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 90%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  gap: 15px;
  color: #2B2C34;
}

.c7 {
  overflow: hidden;
  border-radius: 50%;
  width: 50px;
  height: 50px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 2px;
  gap: 4.2px;
  width: 184px;
  height: 46.2px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

.c9 > svg {
  color: #2B2C34;
  font-size: 2.4rem;
}

.c9 > h1 {
  display: inline;
}

.c0 {
  padding: 4.4rem;
  background-color: #FFFFFE;
}

.c2 {
  margin-top: 1.4rem;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(26rem,2fr));
  gap: 3.576rem;
}

@media (max-width:400px) {
  .c1 {
    font-size: 2.4rem;
  }
}

@media (max-width:400px) {
  .c6 > img {
    width: 47px;
    height: 47px;
  }
}

@media (max-width:400px) {
  .c0 {
    padding: 3.5rem;
  }
}

@media (max-width:1065px) {
  .c2 {
    grid-template-columns: repeat(auto-fill,minmax(27rem,1fr));
  }
}

@media (max-width:705px) {
  .c2 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    grid-template-columns: 0fr;
  }
}

<div
  class="c0"
  id="posts"
>
  <h1
    class="c1"
    color="black"
  >
    Todos os Posts
  </h1>
  <div
    class="c2"
  >
    <article
      class="c3"
      content=""
      id="92632737"
    >
      <img
        alt="image alt"
        data-nimg="1"
        decoding="async"
        height="198"
        loading="lazy"
        src="https://www.datocms-assets.com/90479/1674241450-javascript2.svg"
        srcset="https://www.datocms-assets.com/90479/1674241450-javascript2.svg 1x, https://www.datocms-assets.com/90479/1674241450-javascript2.svg 2x"
        style="color: transparent;"
        width="260"
      />
      <h1
        class="c4"
        color="black"
      >
        Você conhece Typescript ?
      </h1>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="author image"
              data-nimg="1"
              decoding="async"
              height="50"
              src="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
          <div
            class="c8"
          >
            <h4>
              Mateus Duarte
            </h4>
            <p>
              03/02/2023
            </p>
          </div>
        </div>
        <div
          class="c9"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
          <h1
            class="c10"
            color="black"
          >
            0
          </h1>
        </div>
      </div>
    </article>
    <article
      class="c3"
      content=""
      id="92631209"
    >
      <img
        alt="alt image"
        data-nimg="1"
        decoding="async"
        height="198"
        loading="lazy"
        src="https://www.datocms-assets.com/90479/1674241447-javascript1.svg"
        srcset="https://www.datocms-assets.com/90479/1674241447-javascript1.svg 1x, https://www.datocms-assets.com/90479/1674241447-javascript1.svg 2x"
        style="color: transparent;"
        width="260"
      />
      <h1
        class="c4"
        color="black"
      >
        VOCÊ CONHECE JAVASCRIPT ?
      </h1>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="author image"
              data-nimg="1"
              decoding="async"
              height="50"
              src="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
          <div
            class="c8"
          >
            <h4>
              Mateus Duarte
            </h4>
            <p>
              03/02/2023
            </p>
          </div>
        </div>
        <div
          class="c9"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
          <h1
            class="c10"
            color="black"
          >
            0
          </h1>
        </div>
      </div>
    </article>
    <article
      class="c3"
      content=""
      id="95380505"
    >
      <img
        alt="alt image"
        data-nimg="1"
        decoding="async"
        height="198"
        loading="lazy"
        src="https://www.datocms-assets.com/90479/1674241454-javascript4.svg"
        srcset="https://www.datocms-assets.com/90479/1674241454-javascript4.svg 1x, https://www.datocms-assets.com/90479/1674241454-javascript4.svg 2x"
        style="color: transparent;"
        width="260"
      />
      <h1
        class="c4"
        color="black"
      >
        Como usar Array no Javascript
      </h1>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="author image"
              data-nimg="1"
              decoding="async"
              height="50"
              src="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
          <div
            class="c8"
          >
            <h4>
              Mateus Duarte
            </h4>
            <p>
              03/02/2023
            </p>
          </div>
        </div>
        <div
          class="c9"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
          <h1
            class="c10"
            color="black"
          >
            0
          </h1>
        </div>
      </div>
    </article>
    <article
      class="c3"
      content=""
      id="95380585"
    >
      <img
        alt="image alt"
        data-nimg="1"
        decoding="async"
        height="198"
        loading="lazy"
        src="https://www.datocms-assets.com/90479/1674241443-css2.svg"
        srcset="https://www.datocms-assets.com/90479/1674241443-css2.svg 1x, https://www.datocms-assets.com/90479/1674241443-css2.svg 2x"
        style="color: transparent;"
        width="260"
      />
      <h1
        class="c4"
        color="black"
      >
        O que é Tailwind Css ?
      </h1>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="author image"
              data-nimg="1"
              decoding="async"
              height="50"
              src="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
          <div
            class="c8"
          >
            <h4>
              Mateus Duarte
            </h4>
            <p>
              03/02/2023
            </p>
          </div>
        </div>
        <div
          class="c9"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
          <h1
            class="c10"
            color="black"
          >
            0
          </h1>
        </div>
      </div>
    </article>
    <article
      class="c3"
      content=""
      id="95380702"
    >
      <img
        alt="image alt"
        data-nimg="1"
        decoding="async"
        height="198"
        loading="lazy"
        src="https://www.datocms-assets.com/90479/1674241456-react.svg"
        srcset="https://www.datocms-assets.com/90479/1674241456-react.svg 1x, https://www.datocms-assets.com/90479/1674241456-react.svg 2x"
        style="color: transparent;"
        width="260"
      />
      <h1
        class="c4"
        color="black"
      >
        React Hooks: o que é e como funciona ?
      </h1>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="author image"
              data-nimg="1"
              decoding="async"
              height="50"
              src="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
          <div
            class="c8"
          >
            <h4>
              Mateus Duarte
            </h4>
            <p>
              03/02/2023
            </p>
          </div>
        </div>
        <div
          class="c9"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
          <h1
            class="c10"
            color="black"
          >
            0
          </h1>
        </div>
      </div>
    </article>
    <article
      class="c3"
      content=""
      id="95380751"
    >
      <img
        alt="image alt"
        data-nimg="1"
        decoding="async"
        height="198"
        loading="lazy"
        src="https://www.datocms-assets.com/90479/1674243496-vuejs.svg"
        srcset="https://www.datocms-assets.com/90479/1674243496-vuejs.svg 1x, https://www.datocms-assets.com/90479/1674243496-vuejs.svg 2x"
        style="color: transparent;"
        width="260"
      />
      <h1
        class="c4"
        color="black"
      >
        Vue JS: O que é, como funciona ?
      </h1>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="author image"
              data-nimg="1"
              decoding="async"
              height="50"
              src="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
          <div
            class="c8"
          >
            <h4>
              Mateus Duarte
            </h4>
            <p>
              03/02/2023
            </p>
          </div>
        </div>
        <div
          class="c9"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
          <h1
            class="c10"
            color="black"
          >
            0
          </h1>
        </div>
      </div>
    </article>
    <article
      class="c3"
      content=""
      id="92697360"
    >
      <img
        alt="css3"
        data-nimg="1"
        decoding="async"
        height="198"
        loading="lazy"
        src="https://www.datocms-assets.com/90479/1674241439-css.svg"
        srcset="https://www.datocms-assets.com/90479/1674241439-css.svg 1x, https://www.datocms-assets.com/90479/1674241439-css.svg 2x"
        style="color: transparent;"
        width="260"
      />
      <h1
        class="c4"
        color="black"
      >
        o que é CSS3?
      </h1>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="author image"
              data-nimg="1"
              decoding="async"
              height="50"
              src="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
          <div
            class="c8"
          >
            <h4>
              Mateus Duarte
            </h4>
            <p>
              03/02/2023
            </p>
          </div>
        </div>
        <div
          class="c9"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
          <h1
            class="c10"
            color="black"
          >
            0
          </h1>
        </div>
      </div>
    </article>
  </div>
</div>
`;
